### PR TITLE
HBASE-24642: Apache Yetus test-patch and Jenkins integration.

### PR DESCRIPTION
**Pull request by bharathv**
Adds the plumbing needed to run test-patch tooling from Yetus.

Example:

YETUS_ARGS=patch.diff bin/run_tests_with_yetus.sh (without docker)

RUN_IN_DOCKER=true YETUS_ARGS='--dirty-workspace patch.diff' bin/run_tests_with_yetus.sh
(with docker)

Sample output (with a few failing tests).

| Vote |          Subsystem |  Runtime   | Comment
============================================================================
+---------------------------------------------------------------------------
|      |                    |            | Prechecks
+---------------------------------------------------------------------------
+---------------------------------------------------------------------------
|      |                    |            | master Compile Tests
+---------------------------------------------------------------------------
|  +1  |    CMakeLists.txt  |   2m  9s   | master passed
|  +1  |           compile  |   0m 44s   | master passed
+---------------------------------------------------------------------------
|      |                    |            | Patch Compile Tests
+---------------------------------------------------------------------------
|  +1  |    CMakeLists.txt  |   2m  3s   | the patch passed
|  +1  |           compile  |  13m 54s   | the patch passed
+---------------------------------------------------------------------------
|      |                    |            | Other Tests
+---------------------------------------------------------------------------
|  -1  |              unit  |   0m 34s   | root in the patch failed.
|      |                    |  21m  1s   |

              Reason | Tests
 Failed CTEST tests  |  async-batch-rpc-retrying-test
                     |  async-rpc-retrying-test
                     |  client-test
                     |  filter-test
                     |  location-cache-retry-test
                     |  location-cache-test
                     |  scanner-test

Supports both docker and non-docker mode. Runs all the unit-tests
as a part of the test-patch run.

Follow on patches will add the necessary integration with Jenkins
for pre-commit builds.

### CODE REVIEW

**Comment by bharathv** [comment]
@ndimiduk / @busbey Mind taking a look at this one? You folks are probably the most knowledgeable about Yetus related stuff, so appreciate any inputs. To give you some context, @phrocker and I are working on reviving hbase-native-client by fixing the build system and a ton of bugs and as a part of that I'm trying to add a precommit job for testing changes. Effort is tracked in HBASE-24537 and related jiras but I'm happy to fill you in on any additional context that is required. Also if you are interested, please take a look at the recent commits in the repo.  Thanks.

**Comment by bharathv** [comment]
Anyone has some free cycles to take a look at this patch? It should be fairly straight forward with some basic understanding of Yetus. There are some minor quirks with cmake integration but I can add more context as needed. Thank you.

@phrocker FYI. I'm not able to add you as a reviewer, so @ mentioning for heads up.

**Review by bharathv — COMMENTED** [review]
Updated the patch to include the Jenkins integration, its probably easier for reviewers to review them together.

Jenkins job: https://builds.apache.org/view/H-L/view/HBase/job/hbase-native-client-github-precommit/
Sample run (for output): https://github.com/bharathv/hbase-native-client/pull/3 (see comments)

**Comment by phrocker** [comment]
@bharathv I will take a look. sorry for the delay.

**Comment by bharathv** [comment]
@phrocker No worries. How about we get this patch in before #6? That validates both the patches. Tests that the precommit is running well and also tests your patch in a clean environment.

**Review by phrocker — APPROVED** [review]
Lgtm!

File: bin/hbase-native-client-personality.sh [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  local args
+  yetus_info "Personality: ${repostatus} ${testtype}"
+  clear_personality_queue
+  if [[ "${testtype}" =~ CMakeLists.txt ]]; then
+    args=${CMAKE_ARGS}
+    yetus_debug "Appending CMake args ${args}"
+  fi
+  personality_enqueue_module . ${args}
+}
+## This is named so that yetus will check us right after running tests.
```
Comment (joshelser) [inline bin/hbase-native-client-personality.sh]: Whack these comments?

**Review by joshelser — APPROVED** [review]
Looks good, overall. Just some comments to clean up :)

File: bin/hbase-native-client-personality.sh (hunk 1 above)
Comment (bharathv) [reply bin/hbase-native-client-personality.sh]: Done.

**Pull request merged**